mne/preprocessing/tests/test_beer_lambert_law.py

```diff
@@ -0,0 +1,35 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#          Eric Larson <larson.eric.d@gmail.com>
+#          Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#
+# License: BSD (3-clause)
+
+import os.path as op
+
+from mne.datasets.testing import data_path
+from mne.io import read_raw_nirx, BaseRaw
+from mne.preprocessing import optical_density, beer_lambert_law
+from mne.utils import _validate_type
+from mne.datasets import testing
+
+
+fname_nirx = op.join(data_path(download=False),
+                     'NIRx', 'nirx_15_2_recording_w_short')
+
+
+@testing.requires_testing_data
+def test_beer_lambert():
+    """Test return type for optical density."""
+    raw = read_raw_nirx(fname_nirx, preload=False)
+    assert 'fnirs_raw' in raw
+    assert 'fnirs_od' not in raw
+    raw = optical_density(raw)
+    _validate_type(raw, BaseRaw, 'raw')
+    assert 'fnirs_raw' not in raw
+    assert 'fnirs_od' in raw
+    raw = beer_lambert_law(raw)
+    _validate_type(raw, BaseRaw, 'raw')
+    assert 'fnirs_raw' not in raw
+    assert 'fnirs_od' not in raw
+    assert 'hbo' in raw
+    assert 'hbr' in raw
```

Comment:
And eventually it would be great for you to convert the two existing test files to hemoglobin using some other tool, add the result to the `mne-testing-data` dataset, and `assert_allclose` here

Reply:
Something like a script/test version of this?

https://github.com/rob-luke/mne-fnirs-demo/blob/master/BeerLambert_Validation.ipynb

Reply:
Sorry for the slow response, yes something like this where hopefully you end up at `np.allclose` with some suitable tolerance